### DISCUSSION TITLE
Support to supply credentials over file

### DISCUSSION THREAD

**Discussion by sqozz** [librespot-org/librespot#931] [Feature Requests]
I'd love to further secure my librespot setup by supplying the username and password by file or file descriptor to not leak it over the process output. Another alternative would be to supply them over environment variables. As I currently understand this is not possible with the "librespot" daemon. I also tried to search for similar feature request in the bug tracker but didn't find any. Please excuse me if this is a duplicate or I didn't spot the right place in the code where this is already implemented.

**Comment by roderickvd** [comment]
You are right that you can't pass it by pipe.

You can cache the credentials by using `--system-cache` option. This works both with zeroconf discovery and with username on the command line. Note: I've never tried but I'm not sure if you can pipe the password if you only specify `--user`.

Though I can understand your request, I think it's a pretty edge case amongst our users and not something I will pursue myself. However, I will consider any PR's for it.

**Comment by kingosticks** [comment]
>Another alternative would be to supply them over environment variables. As I currently understand this is not possible with the "librespot" daemon 

Support for environment variables provided thanks to https://github.com/librespot-org/librespot/pull/886

**Reply by roderickvd** [reply]
I should've remembered that also worked for username and password 🙈 thanks.

**Reply by roderickvd** [reply]
@JasonLG1979 come to think of it, I think we forgot to update the wiki?
